include/mbgl/util/position.hpp

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <mbgl/style/types.hpp>
+#include <mbgl/util/optional.hpp>
+#include <mbgl/util/constants.hpp>
+
+#include <array>
+
+#include <cassert>
+#include <string>
+
+namespace mbgl {
+namespace style {
+class Position {
+public:
+    Position() = default;
+    Position(std::array<float, 3> position_)
+        : radial(position_[0]), azimuthal(position_[1]), polar(position_[2]) {
+        calculateCartesian();
+    };
+
+    friend bool operator==(const Position& lhs, const Position& rhs) {
+        return lhs.radial == rhs.radial && lhs.azimuthal == rhs.azimuthal && lhs.polar == rhs.polar;
+        // TODO should this address wrapping? or should somewhere else in Position address wrapping?
+    }
+
+    friend bool operator!=(const Position& lhs, const Position& rhs) {
+        return !(lhs == rhs);
+    }
+
+    const std::array<float, 3> get() const {
+        return { { x, y, z } };
+    };
+
+    const std::array<float, 3> getSpherical() const {
+        return { { radial, azimuthal, polar } };
+    };
+
+    void set(std::array<float, 3> position_) {
+        radial = position_[0];
+        azimuthal = position_[1];
+        polar = position_[2];
+        calculateCartesian();
+    };
+
+private:
+    float radial;
+    float azimuthal;
+    float polar;
+    float x;
+    float y;
+    float z;
+
+    void calculateCartesian() {
+        auto _a = azimuthal;
```

Comment:
Write the calculations as a one liner, e.g. `const float _a = (azimuthal + 90) * util::DEG2RAD`.